[#90060190] Allow access to the microphone data. You can use it to view/...

promanik opened the pull request:
...visualize it, or write the data to the output to be heard.

SpecialDOS commented on classes/microphone.dre:
```diff
@@ -24,10 +24,34 @@ SOFTWARE. -->
 <class name="microphone" type="coffee" clickable="true" clip="true" scriptincludes="/lib/audiorecorder/js/recorderjs/recorder.js, /lib/filesaver/FileSaver.min.js">
   
   <!--/**
+    * @attribute {Boolean} [visualize=true]
```
I think the doc comment for this is misleading, a visualization will not be automatically be displayed, but the fft data will be available for generating a display

SpecialDOS commented on classes/microphone.dre:
```diff
@@ -147,14 +200,29 @@ SOFTWARE. -->
-  
+
+      # FFT visualization
+      if @visualize
```
It looks like a side affect of this conditional is that recording time will only be set if @visualize is true, because the onidle handler returns early unless @_fftNode is set, so I think you'll need to adjust the logic in onidle a bit

SpecialDOS commented on examples/rtmicrophone.html:
```diff
@@ -0,0 +1,107 @@
+<html>
```
It might make sense to roll this processStream example into the existing example examples/microphone.html, since that one does the same things otherwise as this example does. But that example is a little broken right now because the control icons are not showing up, which looks like an issue with the latest dreem text changes, and I'm looking into it now.

promanik commented:
Lorien, I changed the rtmicrophone app to display real-time feedback of the audio amplitude. I removed the fft visualization because it duplicated the functionality in microphone.html.
